### PR TITLE
chore: downstream check for all libraries in single job

### PR DESCRIPTION
**Pull request by Neenu1995**
Test core java libraries in java client libraries before release. Feature/code changes introduced in each PR can be tested on downstream java client libraries by adding `downstream-check:run` label. For now, unit tests are being run.

Next phase involves running samples and integration tests.

Note: Some of these tests are expected to fail due to individual client library conditions and do block the merging of this PR.

### CODE REVIEW

File: .kokoro/downstream-client-library-check.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+REPO=$1
+## Get the directory of the build script
+scriptDir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+## cd to the parent directory, i.e. the root of the git repo
+cd ${scriptDir}/..
+
+# Make java-core artifacts available for 'mvn validate' at the bottom
+mvn install -DskipTests=true -Dmaven.javadoc.skip=true -Dgcloud.download.skip=true -B -V -q
+
+# Read the current version of this java-core in the POM. Example version: '0.116.1-alpha-SNAPSHOT'
```
Comment (suztomo) [inline .kokoro/downstream-client-library-check.sh:33]: I guess this is not java-core any more.

File: .kokoro/downstream-client-library-check.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+AUTH_VERSION_POM=pom.xml
+# Namespace (xmlns) prevents xmllint from specifying tag names in XPath
+AUTH_VERSION=`sed -e 's/xmlns=".*"//' ${AUTH_VERSION_POM} | xmllint --xpath '/project/version/text()' -`
```
Comment (suztomo) [inline .kokoro/downstream-client-library-check.sh:36]: As of now, I see this script is specific to auth library (from the variable name "AUTH_VERSION)). But I believe this script is beneficial to Google HTTP client, GAX, and all other libraries in the shared dependencies BOM. What do you think about making it generic?

**Review by suztomo — COMMENTED** [review]
It's nice to see the 132 checks ran.

File: .kokoro/downstream-client-library-check.sh (hunk 2 above)
Comment (Neenu1995) [reply .kokoro/downstream-client-library-check.sh:36]: I will update it to be more generic.

File: .kokoro/downstream-client-library-check.sh (hunk 1 above)
Comment (Neenu1995) [reply .kokoro/downstream-client-library-check.sh]: Fixed

**Review by lesv — APPROVED** [review]

**Review by lesv — APPROVED** [review]

**Pull request merged**